src/main/java/org/terasology/dynamicCities/settlements/SettlementEntityManager.java

```diff
@@ -219,11 +220,10 @@ public void checkZoneNeeded(CheckZoneNeededEvent event, EntityRef settlement, Cu
 
     public boolean checkMinDistance(EntityRef siteRegion) {
         Vector3fc sitePos = siteRegion.getComponent(LocationComponent.class).getLocalPosition();
-        Vector2i pos = new Vector2i(sitePos.x(), sitePos.z());
+        Vector2i pos = new Vector2i(sitePos.x(), sitePos.z(), RoundingMode.FLOOR);
         SettlementsCacheComponent container = settlementCachingSystem.getSettlementCacheEntity().getComponent(SettlementsCacheComponent.class);
-        for (String vector2iString : container.settlementEntities.keySet()) {
-            Vector2i activePosition = Toolbox.stringToVector2i(vector2iString);
-            if (pos.distance(activePosition) < minDistance) {
+        for (Vector2i loc : container.settlementEntities.keySet()) {
+            if (pos.distance(loc) < minDistance) {
```

Comment:
Here, we are comparing the `.getLocalPosition()` of the `siteRegion` to the world position of the existing settlements... See https://github.com/Terasology/DynamicCities/pull/86/files?file-filters%5B%5D=.java&hide-deleted-files=true#diff-cfa75a8f8216b299e8adccccdcd42868081946cd829cb8b4a837ec3eddb50e04R39

Probably irrelevant since we don't have parents with locations for settlements (or do we?).